### PR TITLE
geth/v1.14.10 upstream merge

### PR DESCRIPTION
**Pull request by 0x00101010**
**Description**

Merges in upstream geth/v1.14.10

**Additional context**

N/A

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,6 @@
 module github.com/ethereum/go-ethereum
 
-go 1.21
-
-toolchain go1.21.6
+go 1.22
```
Comment (mdehoog) [inline go.mod:3]: @0x00101010 see this comment from @ajsutton regarding Go 1.22: https://github.com/ethereum-optimism/op-geth/pull/389#issuecomment-2375606462

You can switch back to 1.21 by downgrading the go-verkle library as done in https://github.com/ethereum-optimism/op-geth/pull/389

Comment (0x00101010) [reply go.mod:3]: Thanks @mdehoog 

1. You can close your PR now as upgrade to [v1.14.10](https://github.com/ethereum-optimism/optimism/pull/12197) covers it, I'm in contact with Seb and team to get it merged
2. Regarding golang v1.22, it is fixed now and we're good to go ahead: https://github.com/ethereum-optimism/optimism/pull/12197#issuecomment-2381653018, https://github.com/ethereum-optimism/optimism/pull/12197#issuecomment-2384168799

Comment (mdehoog) [reply go.mod:3]: oh nice 🙌

File: build/ci.go [hunk 2]
```diff
@@ -254,7 +256,11 @@ func buildFlags(env build.Environment, staticLinking bool, buildTags []string) (
 	if runtime.GOOS == "linux" {
 		// Enforce the stacksize to 8M, which is the case on most platforms apart from
 		// alpine Linux.
-		extld := []string{"-Wl,-z,stack-size=0x800000"}
+		// See https://sourceware.org/binutils/docs-2.23.1/ld/Options.html#Options
+		// regarding the options --build-id=none and --strip-all. It is needed for
+		// reproducible builds; removing references to temporary files in C-land, and
+		// making build-id reproducably absent.
+		extld := []string{"-Wl,-z,stack-size=0x800000,--build-id=none,--strip-all"}
```
Comment (ajsutton) [inline build/ci.go:263]: Note: We need to check that op-program builds are still reproducible after this update. I'm not sure exactly what's going on with these options but if they're needed for geth builds to be reproducible we may need them for op-program too.

File: core/rawdb/freezer.go [hunk 3]
```diff
@@ -389,115 +385,3 @@ func (f *Freezer) repair() error {
...
 	return nil
 }
-
-// convertLegacyFn takes a raw freezer entry in an older format and
-// returns it in the new format.
-type convertLegacyFn = func([]byte) ([]byte, error)
-
-// MigrateTable processes the entries in a given table in sequence
-// converting them to a new format if they're of an old format.
-func (f *Freezer) MigrateTable(kind string, convert convertLegacyFn) error {
```
Comment (ajsutton) [inline core/rawdb/freezer.go:399]: This is removing some data migration. We should double check the migration has existing in op-geth long enough to ensure everyone has migrated.

**Review by ajsutton — APPROVED** [review]
LGTM.  Seems like a pretty straight forward upgrade for us.  Just a couple of notes of things to double check.

File: build/ci.go (hunk 2 above)
Comment (ajsutton) [reply build/ci.go:263]: Could you run `make reproducible-prestate`  in the monorepo with https://github.com/ethereum-optimism/optimism/pull/12197 checked out please?  It should end with:
```
Cannon Absolute prestate hash:
0x03f7fb0a1cfc0c300e91c9cf83af2478c954efa1ab6fa4703b5ec6011797be2b
MT-Cannon Absolute prestate hash:
0x0370379f6a181d8c5c44aad4af3e0868a37437205653fbccb1fd6e662e62fac8
```

Comment (0x00101010) [reply build/ci.go:263]:
```
Cannon Absolute prestate hash:
0x03f7fb0a1cfc0c300e91c9cf83af2478c954efa1ab6fa4703b5ec6011797be2b
MT-Cannon Absolute prestate hash:
0x0370379f6a181d8c5c44aad4af3e0868a37437205653fbccb1fd6e662e62fac8
```

Comment (0x00101010) [reply build/ci.go:263]: Great, seems like they match!

File: core/rawdb/freezer.go (hunk 3 above)
Comment (0x00101010) [reply core/rawdb/freezer.go:399]: I believe so, last edits to this line / related functions is 2 years ago:
image

**Review by saeidazimi512 — APPROVED** [review]

**Pull request closed without merging**